packages/go_router/lib/src/route.dart

```diff
@@ -1218,6 +1218,18 @@ class StatefulNavigationShell extends StatefulWidget {
     assert(index >= 0);
     return index;
   }
+
+  /// trigger an update of [_branchState] to make dynamically added
+  /// [StatefulShellBranch]es available
+  void updateBranches() {
```

Comment:
I think you should be able to achieve the same thing with GoRouter.routerConfig().